mrbgem.rake

```diff
@@ -2,7 +2,47 @@ MRuby::Gem::Specification.new('mruby-redis') do |spec|
   spec.license = 'MIT'
   spec.authors = 'MATSUMOTO Ryosuke'
   spec.version = '0.0.1'
-  spec.linker.libraries << "hiredis"
   # for expire test
+  require 'open3'
+
+  hiredis_dir = "#{build_dir}/hiredis"
+  spec.linker.libraries << "hiredis"
+
+  def run_command env, command
+    STDOUT.sync = true
+    puts "build: [exec] #{command}"
+    Open3.popen2e(env, command) do |stdin, stdout, thread|
+      print stdout.read
+      fail "#{command} failed" if thread.value != 0
+    end
+  end
+
+  FileUtils.mkdir_p build_dir
+
+  if ! File.exists? hiredis_dir
+    Dir.chdir(build_dir) do
+      e = {}
+      run_command e, 'git clone git://github.com/redis/hiredis.git'
+    end
+  end
+
+  if ! File.exists? "#{hiredis_dir}/libhiredis.a"
+    Dir.chdir hiredis_dir do
+      e = {
+        'CC' => "#{spec.build.cc.command} #{spec.build.cc.flags.join(' ')}",
+        'CXX' => "#{spec.build.cxx.command} #{spec.build.cxx.flags.join(' ')}",
+        'LD' => "#{spec.build.linker.command} #{spec.build.linker.flags.join(' ')}",
+        'AR' => spec.build.archiver.command,
+        'PREFIX' => hiredis_dir
+      }
+
+      run_command e, "make"
+      run_command e, "make install"
+    end
+  end
+
+  spec.cc.include_paths << hiredis_dir
+  spec.linker.library_paths << hiredis_dir
+
```

Comment:
If we use libhiredis.a for static link, `spec.cc` and `spec.linker` flag is the following configuration:

```
  spec.cc.include_paths << "#{hiredis_dir}/include"                              
  #spec.linker.library_paths << hiredis_dir                                      
  spec.linker.flags_before_libraries << "#{hiredis_dir}/lib/libhiredis.a"        
```